packages/react/src/primitives/TextArea/__tests__/TextArea.test.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { TextArea } from '../TextArea';
+import { ComponentClassNames } from '../../shared';
+
+describe('TextArea component', () => {
+  it('should render custom classname for TextArea', async () => {
+    render(<TextArea className="custom-class" />);
+
+    const textarea = (await screen.findByRole(
+      'textbox'
+    )) as HTMLTextAreaElement;
+    expect(textarea).toHaveClass('custom-class');
+    expect(textarea).toHaveClass(ComponentClassNames.Textarea);
+  });
+
+  it('should render expected classname, id TextArea field', async () => {
+    render(
+      <TextArea
+        id="testField"
+        testId="testId"
+        className="my-textarea"
+        defaultValue="Hello there"
+      />
+    );
+
+    const textarea = await screen.findByRole('textbox');
+    expect(textarea).toHaveClass('my-textarea');
+    expect(textarea.id).toBe('testField');
+  });
+
+  it('should forward ref to DOM element', async () => {
+    const ref = React.createRef<HTMLTextAreaElement>();
+
+    render(<TextArea ref={ref} />);
+    await screen.findByRole('textbox');
+    expect(ref.current.nodeName).toBe('TEXTAREA');
+  });
+
+  it('should render the state attributes', async () => {
+    render(<TextArea size="small" hasError isDisabled isReadOnly isRequired />);
+
+    const textarea = await screen.findByRole('textbox');
+    expect(textarea).toHaveAttribute('disabled');
+    expect(textarea).toHaveAttribute('readonly');
+    expect(textarea).toHaveAttribute('required');
```

Comment:
nit: Do you mind updating those to test against their real boolean values? This makes me feel a bit nervous anyway

Reply:
+1

Reply:
Sure, but when true, the underlying DOM attribute will be an empty string, not a boolean. Updated to specifically target empty string.